pkg/store/multitsdb.go

```diff
@@ -0,0 +1,156 @@
+package store
+
+import (
+	"context"
+	"sort"
+
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/thanos-io/thanos/pkg/component"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+)
+
+type MultiTSDBStore struct {
+	logger     log.Logger
+	component  component.SourceStoreAPI
+	tsdbStores func() []*TSDBStore
+}
+
+// NewMultiTSDBStore creates a new TSDBStore.
+func NewMultiTSDBStore(logger log.Logger, _ prometheus.Registerer, component component.SourceStoreAPI, tsdbStores func() []*TSDBStore) *MultiTSDBStore {
+	if logger == nil {
+		logger = log.NewNopLogger()
+	}
+	return &MultiTSDBStore{
+		logger:     logger,
+		component:  component,
+		tsdbStores: tsdbStores,
+	}
+}
+
+// Info returns store information about the Prometheus instance.
+func (s *MultiTSDBStore) Info(ctx context.Context, req *storepb.InfoRequest) (*storepb.InfoResponse, error) {
+	stores := s.tsdbStores()
+
+	resp := &storepb.InfoResponse{
+		StoreType: s.component.ToProto(),
+	}
+	if len(stores) == 0 {
+		return resp, nil
+	}
+
+	infos := make([]*storepb.InfoResponse, 0, len(stores))
+	for _, store := range stores {
+		info, err := store.Info(ctx, req)
+		if err != nil {
+			return nil, err
+		}
+		infos = append(infos, info)
+	}
+
+	resp.MinTime = infos[0].MinTime
+	resp.MaxTime = infos[0].MaxTime
+
+	for i := 1; i < len(infos); i++ {
+		if resp.MinTime > infos[i].MinTime {
+			resp.MinTime = infos[i].MinTime
+		}
+		if resp.MaxTime < infos[i].MaxTime {
+			resp.MaxTime = infos[i].MaxTime
+		}
+	}
+
+	seenLabelSets := map[uint64]struct{}{}
+	labelSets := []storepb.LabelSet{}
+	for _, info := range infos {
+		for _, labelSet := range info.LabelSets {
+			ls := storepb.LabelsToPromLabels(labelSet.Labels)
+			sort.Sort(ls)
+			h := ls.Hash()
+			_, seen := seenLabelSets[h]
+			if !seen {
+				seenLabelSets[h] = struct{}{}
+				labelSets = append(labelSets, labelSet)
+			}
+		}
+	}
+	resp.LabelSets = labelSets
+
+	return resp, nil
+}
+
+// Series returns all series for a requested time range and label matcher. The returned data may
+// exceed the requested time bounds.
+func (s *MultiTSDBStore) Series(r *storepb.SeriesRequest, srv storepb.Store_SeriesServer) error {
+	stores := s.tsdbStores()
+	for _, store := range stores {
+		err := store.Series(r, srv)
```

Comment:
`if err := ...; err != nil { ` pattern is available here

Reply:
Hm..... does this works? Don't we need some proxyStore here?

Reply:
Picking this up now.